crates/matrix-sdk-base/src/store/mod.rs

```diff
@@ -380,6 +388,35 @@ pub trait StateStore: AsyncTraitDeps {
 #[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
 pub trait StateStoreExt: StateStore {
+    /// Get session data from the store.
+    ///
+    /// # Arguments
+    ///
+    /// * `key` - The key to fetch data for
+    async fn get_session_data<T: DeserializeOwned>(
+        &self,
+        key: SessionDataKey<'_>,
+    ) -> Result<Option<T>> {
+        let value_bytes = self.get_raw_session_data(key).await?;
+        value_bytes.as_deref().map(serde_json::from_slice).transpose().map_err(Into::into)
+    }
+
+    /// Put session data into the store.
+    ///
+    /// # Arguments
+    ///
+    /// * `key` - The key to insert data for
+    ///
+    /// * `value` - The value to insert
+    async fn set_session_data(
+        &self,
+        key: SessionDataKey<'_>,
+        value: &(impl Serialize + Send + Sync),
+    ) -> Result<()> {
```

Comment:
I remember @poljar telling that the idea of using the StateStore for the custom value is that it should behave a bit like the `BTreeMap`, so the set/insert of this specific key and value should return an optional value that corresponds to the value that was present before the change (if available).
However I am not sure if that is really needed honestly, I actually prefer it to return an empty Result.

Reply:
I'm not sure it's really necessary when the API is only supposed to be used internally and we don't need it.